app/javascript/components/subject/CaseStatus.js

```diff
@@ -31,42 +31,41 @@ class CaseStatus extends React.Component {
 
   handleChange(event) {
     event.persist();
-
     const value = event.target.type === 'checkbox' ? event.target.checked : event.target.value;
-    const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
-    const hideModal = this.state.isolation && confirmedOrProbable;
 
-    this.setState({ [event.target.id]: value, showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
-      // specific case where case status is just changed with no modal
-      if (hideModal) {
-        this.setState({ message: 'case status to "' + this.state.case_status + '".' });
-        this.submit();
-      }
+    this.setState({ [event.target.id]: value }, () => {
+      if (event.target.id === 'case_status') {
+        const confirmedOrProbable = value === 'Confirmed' || value === 'Probable';
+        const hideModal = this.state.isolation && confirmedOrProbable;
 
-      // all other cases
-      if (event.target.id === 'confirmed') {
+        this.setState({ showCaseStatusModal: !hideModal, confirmedOrProbable }, () => {
+          // specific case where case status is just changed with no modal
+          if (hideModal) {
+            this.setState({ message: 'case status to "' + this.state.case_status + '".' });
```

Comment:
We just merged Tim's PR which refactored the History logic to be entirely on the backend, so this will need to be updated.

Reply:
all set